Game/RainMeadow.PlayerHooks.cs

```diff
@@ -301,6 +304,65 @@ private Player.ObjectGrabability PlayerOnGrabability(On.Player.orig_Grabability
         return orig(self, obj);
     }
 
+    private void Player_SwallowObject(ILContext il)
+    {
+        try
+        {
+            var c = new ILCursor(il);
+            var skip = il.DefineLabel();
+            c.GotoNext(moveType: MoveType.After,
+                i => i.MatchLdarg(0),
+                i => i.MatchLdloc(0),
+                i => i.MatchStfld<Player>("objectInStomach")
+                );
+            c.Emit(OpCodes.Ldarg_0);
+            c.EmitDelegate((Player self) =>
+            {
+                if (OnlineManager.lobby != null && OnlinePhysicalObject.map.TryGetValue(self.abstractPhysicalObject, out var oe))
+                {
+                    if (!oe.isMine) return false;
+                    if (OnlinePhysicalObject.map.TryGetValue(self.objectInStomach, out var oeInStomach))
+                        oeInStomach.isTransferable = false; // don't release ownership
```

Comment:
This field of OnlineEntity isn't synced, the player that sets it will have a unique perspective about it.

Might still work for the sake of controlling whether to release or to accept requests. However "region owner" might still transfer it if asked directly. Clients ask the region owner if the entity is in a resource the direct owner isn't it (assumed abandoned).

Reply:
huuh.
should we want to sync isTransferable too? or use that shouldRealize bool
(I'm not too happy with the original "solution" of realizing first then checking if it's a stomach object)